derive/src/lib.rs

```diff
@@ -23,7 +32,43 @@ macro_rules! ident {
     };
 }
 
-fn expand_contract(input: LitStr) -> Result<TokenStream> {
+struct ContractArgs {
+    artifact_path: LitStr,
+    runtime_crate: Option<Ident>,
+}
+
+impl Parse for ContractArgs {
+    fn parse(input: ParseStream) -> ParseResult<Self> {
+        let mut result = ContractArgs {
+            artifact_path: input.parse()?,
+            runtime_crate: None,
+        };
+
+        while !input.is_empty() {
+            input.parse::<Token![,]>()?;
+            if input.is_empty() {
+                break;
```

Comment:
Should we break here or also return an error?

Reply:
breaking here allows for trailing commas. So:
```rust
contract!(
    "path/to/artifact.json",
    foo = bar,
    baz = baw,
);
```
should work.